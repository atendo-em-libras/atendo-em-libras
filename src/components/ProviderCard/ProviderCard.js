import React, { useState } from 'react'
import {
  SectionTitle,
  License,
  SectionBlock,
  ExerienceText,
  ProviderName,
  ProviderVideoCallAvailabilityContainer,
  ProviderVideoCallAvailability,
  ProviderSpecialty,
  ProviderCardWrapper,
} from './ProviderCardStyles'
import { IconText } from '../IconText'
import PropTypes from 'prop-types'
import { CategoryBadge } from './CategoryBadge'
import { Paragraph, Box, ResponsiveContext, Collapsible } from 'grommet'
import { FormDown, FormUp } from 'grommet-icons'
import { videoIcon, phoneIcon, emailIcon, addressIcon, healthInsuranceIcon } from '../../assets/icons'
import { ResponsiveGrid } from '../ResponsiveGrid'
import { ToggleButton } from './ToggleButton'

const ProviderCard = (props) => {
  const [open, setOpen] = useState(false)

  const columns = {
    small: ['auto'],
    medium: ['1/2', '1/2'],
    large: ['1/2', '1/2'],
    xlarge: ['1/2', '1/2'],
  }

  const rows = {
    small: ['auto', 'auto', 'auto'],
    medium: ['auto', 'auto'],
    large: ['auto', 'auto'],
    xlarge: ['auto', 'auto'],
  }

  const fixedGridAreas = {
    small: [
      { name: 'header', start: [0, 0], end: [0, 0] },
      { name: 'first', start: [0, 1], end: [0, 1] },
      { name: 'second', start: [0, 2], end: [0, 2] },
    ],
    medium: [
      { name: 'header', start: [0, 0], end: [1, 0] },
      { name: 'first', start: [0, 1], end: [0, 1] },
      { name: 'second', start: [1, 1], end: [1, 1] },
    ],
    large: [
      { name: 'header', start: [0, 0], end: [1, 0] },
      { name: 'first', start: [0, 1], end: [0, 1] },
      { name: 'second', start: [1, 1], end: [1, 1] },
    ],
    xlarge: [
      { name: 'header', start: [0, 0], end: [1, 0] },
      { name: 'first', start: [0, 1], end: [0, 1] },
      { name: 'second', start: [1, 1], end: [1, 1] },
    ],
  }

  const {
    name,
    category,
    specialty,
    licenseNumber,
    videoCallAvailability,
    phoneNumber,
    email,
    experience,
    healthInsurance,
    city,
    videoCallPlatform,
    state,
    address,
  } = props.provider

  const ProviderVideoCallAvailabilityLabel = () => (
    <ProviderVideoCallAvailabilityContainer>
      <ProviderVideoCallAvailability role="note">
        <IconText src={videoIcon} alt="video icon" />
        Atende Online
      </ProviderVideoCallAvailability>
    </ProviderVideoCallAvailabilityContainer>
  )

  const DetailsInfo = () => (
    <>
      <ProviderName>{name}</ProviderName>
      <ProviderSpecialty>{specialty}</ProviderSpecialty>
      <License>
        <p className="title">REGISTRO</p>
        {licenseNumber ? <p className="number">{licenseNumber}</p> : <p className="number">-</p>}
      </License>
      {experience && <ExerienceText size="small">{experience}</ExerienceText>}
      <SectionBlock>
        {state && (
          <SectionTitle>
            <IconText src={addressIcon} alt="address icon" />
            {city && `${city} - `}
            {`${state}`}
          </SectionTitle>
        )}
        <Paragraph size="small">{address}</Paragraph>
      </SectionBlock>
    </>
  )

  const ContactInfo = () => (
    <SectionBlock>
      <SectionTitle>Contato</SectionTitle>
      <p>
        <IconText src={phoneIcon} alt="phone icon" />
        <span> {phoneNumber}</span>
      </p>
      <Paragraph size="small">
        <IconText src={emailIcon} alt="e-mail icon" />
        <span> {email}</span>
      </Paragraph>
    </SectionBlock>
  )

  const CallPlatform = () => (
    <SectionBlock>
      <SectionTitle>Atendimento online</SectionTitle>
      <p>
        <IconText src={videoIcon} alt="video icon" />
        {videoCallPlatform ? <span>{videoCallPlatform}</span> : <span>Não informado</span>}
      </p>
    </SectionBlock>
  )

  const HealthInsurance = () => (
    <SectionBlock>
      <SectionTitle>Planos de saúde</SectionTitle>
      <p>
        <IconText src={healthInsuranceIcon} alt="card icon" />
        {healthInsurance ? <span>{healthInsurance}</span> : <span>Não informado</span>}
      </p>
    </SectionBlock>
  )

  const MoreInfoBox = () => (
    <>
      {videoCallAvailability && <CallPlatform />}
      <HealthInsurance />
    </>
  )

  return (
    <ProviderCardWrapper className="providerCard" role="listitem">
      <ResponsiveGrid columns={columns} rows={rows} areas={fixedGridAreas}>
        <Box gridArea="header">
          {videoCallAvailability && <ProviderVideoCallAvailabilityLabel />}
          <CategoryBadge category={category} />
        </Box>

        <ResponsiveContext.Consumer>
          {(responsive) =>
            responsive === 'small' ? (
              <Box gridArea="first" className="providerCardColumn">
                <DetailsInfo />
                <ContactInfo />
                <Collapsible open={open} {...props}>
                  <MoreInfoBox />
                </Collapsible>
                <ToggleButton
                  isOpen={open}
                  labelClose="Fechar"
                  labelOpen="Saiba Mais"
                  iconClose={FormUp}
                  iconOpen={FormDown}
                  onClick={() => {
                    setOpen(!open)
                  }}
                />
              </Box>
            ) : (
              <>
                <Box gridArea="first" className="providerCardColumn">
                  <DetailsInfo />
                </Box>
                <Box gridArea="second" className="providerCardColumn">
                  <ContactInfo />
                  <MoreInfoBox />
                </Box>
              </>
            )
          }
        </ResponsiveContext.Consumer>
      </ResponsiveGrid>
    </ProviderCardWrapper>
  )
}

ProviderCard.propTypes = {
  provider: PropTypes.shape({
    name: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    specialty: PropTypes.string.isRequired,
    licenseNumber: PropTypes.string.isRequired,
    videoCallAvailability: PropTypes.bool,
    phoneNumber: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    experience: PropTypes.string,
    healthInsurance: PropTypes.string,
    videoCallPlatform: PropTypes.string,
    city: PropTypes.string,
    state: PropTypes.string,
    address: PropTypes.string,
  }).isRequired,
}

export { ProviderCard }
