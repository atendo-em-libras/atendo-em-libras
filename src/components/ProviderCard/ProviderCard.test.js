import React from 'react'
import { ProviderCard } from './ProviderCard'
import { render, screen, fireEvent } from '@testing-library/react'
import mockProviders from '../../mocks/providers'
import { ResponsiveContext } from 'grommet'
import 'jest-styled-components'

describe('ProviderCard tests', () => {
  let view

  const provider = mockProviders.providers[0]

  beforeEach(() => {
    view = render(<ProviderCard provider={provider} />)
  })

  it('Component should render name property', () => {
    expect(screen.getByText(provider.name)).toBeInTheDocument()
  })

  it('Component should render category property', () => {
    expect(screen.getByText(provider.category)).toBeInTheDocument()
  })

  it('Component should render specialty property', () => {
    expect(screen.getByText(provider.specialty)).toBeInTheDocument()
  })

  it('Component should render license number', () => {
    expect(screen.getByText(provider.registerNumber)).toBeInTheDocument()
  })

  it('Component should render phone number', () => {
    expect(screen.getByText(provider.phone)).toBeInTheDocument()
  })

  it('Component should render email', () => {
    expect(screen.getByText(provider.email)).toBeInTheDocument()
  })

  it('Component should render experience', () => {
    expect(screen.getByText(provider.biography)).toBeInTheDocument()
  })

  it('Component should render health insurence', () => {
    expect(screen.getByText(provider.healthInsurance)).toBeInTheDocument()
  })

  it('Component should render video call platform', () => {
    provider.attendance.onlineAttendance = {
      platforms: 'Zoom',
    }

    view.rerender(<ProviderCard provider={provider} />)

    expect(screen.getByText(provider.attendance.onlineAttendance.platforms)).toBeInTheDocument()
  })

  it('Component should render city and state', () => {
    console.log(screen)
    expect(
      screen.getByText(
        `${provider.attendance.hospitalClinicAttendance.city} - ${provider.attendance.hospitalClinicAttendance.state} (${provider.attendance.hospitalClinicAttendance.stateInitials})`
      )
    ).toBeInTheDocument()
  })

  it('Component should render only state when city is not informed', () => {
    provider.attendance.hospitalClinicAttendance.city = ''

    view.rerender(<ProviderCard provider={provider} />)

    expect(screen.getByText(`${provider.attendance.hospitalClinicAttendance.state}`)).toBeInTheDocument()
  })

  it('Component should render address', () => {
    expect(screen.getByText(provider.attendance.hospitalClinicAttendance.streetName)).toBeInTheDocument()
  })

  it('Component should render video call avallability', () => {
    view.rerender(
      <ProviderCard provider={{ ...provider, attendance: { ...provider.attendance, onlineAttendance: {} } }} />
    )

    expect(screen.getByText('Atende Online')).toBeInTheDocument()
  })

  it('Component should not render video call avallability', () => {
    view.rerender(
      <ProviderCard provider={{ ...provider, attendance: { ...provider.attendance, onlineAttendance: null } }} />
    )

    expect(screen.queryByText('Atende Online')).toBeNull()
  })

  it('Component should render correctly', () => {
    const { container } = view
    expect(container).toMatchSnapshot()
  })

  describe('when mobile', () => {
    it('Component should render correctly', () => {
      view.rerender(
        <ResponsiveContext.Provider value={'small'}>
          <ProviderCard provider={provider} />
        </ResponsiveContext.Provider>
      )
      const { container } = view
      expect(container).toMatchSnapshot()
    })

    it('Should show and hide more info', () => {
      view.rerender(
        <ResponsiveContext.Provider value={'small'}>
          <ProviderCard provider={provider} />
        </ResponsiveContext.Provider>
      )

      expect(screen.getByText('Saiba Mais')).toBeInTheDocument()

      const buttonSaibaMais = screen.getByRole('button')
      fireEvent.click(buttonSaibaMais)
      expect(screen.getByText('Fechar')).toBeInTheDocument()

      const buttonFechar = screen.getByRole('button')
      fireEvent.click(buttonFechar)
      expect(screen.getByText('Saiba Mais')).toBeInTheDocument()
    })
  })
})
