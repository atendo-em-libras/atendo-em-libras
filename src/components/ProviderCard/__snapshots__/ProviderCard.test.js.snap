// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderCard tests Component should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: header;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: first;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: second;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "header header" "first second";
  grid-template-columns: 50% 50%;
  grid-row-gap: 0px;
  grid-column-gap: 0px;
  justify-items: stretch;
  grid-template-rows: auto auto;
}

.c15 {
  font-size: 14px;
  line-height: 20px;
  max-width: 336px;
}

.c1 {
  font-size: 10px;
  border-radius: 15px;
  box-shadow: 0px 10px 32px #0000001a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
  overflow: hidden;
  padding: 2em;
  line-height: 3em;
  word-break: break-word;
}

.c1 .providerCardColumn:last-child {
  border-right: none;
  margin-right: none;
}

.c10 {
  color: #0f1423;
  font-size: 1.4em;
  max-width: 30em;
  line-height: 20px;
  margin-top: 1em;
}

.c4 {
  width: 100%;
}

.c5 {
  color: #5996f7;
  background-color: #c7ddfd;
  border-radius: 0px 15px;
  font-size: 1.4em;
  font-weight: 600;
  padding: 0.5em 1em;
  position: absolute;
  right: 0;
  top: 0;
}

.c9 {
  font-weight: 600;
  color: #0f1423;
  font-size: 1.8em;
}

.c13 {
  border-bottom: #f5f5f5 1.5px solid;
  padding-bottom: 2em;
  margin-bottom: 2em;
  font-size: 1.4em;
}

.c13:last-child {
  border-bottom: none;
  padding-bottom: 0;
  margin-bottom: 0;
}

.c14 {
  font-weight: 600;
  color: #0f1423;
}

.c11 {
  margin-bottom: 1.5em;
  margin-top: 1.5em;
  line-height: 20px;
}

.c11 .title {
  font-weight: 800;
  color: #5996f7;
  font-size: 1.2em;
}

.c11 .number {
  color: #0f1423;
  font-size: 1.4em;
}

.c12 {
  font-size: 1.3em;
  line-height: 1.5em;
  color: #0f1423;
  opacity: 0.57;
}

.c6 {
  vertical-align: middle;
  margin-right: 5px;
}

.c7 {
  margin-right: 1em;
  color: #fff;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 70%;
  font-size: 1.4em;
  font-weight: 600;
  padding: 0.2em 2em;
  border-radius: 1.5em;
  line-height: 1.8em;
  background-color: #ED985A;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1.5em;
}

@media (min-width:800px) {
  .c1 .providerCardColumn {
    border-right: #f5f5f5 1.5px dashed;
    padding-right: 3em;
  }

  .c1 .providerCardColumn:last-child {
    padding-left: 3em;
  }
}

@media (max-width:800px) {
  .c1 {
    padding-bottom: 10em;
  }
}

@media (max-width:440px) {
  .c4 {
    margin-top: 40px;
  }
}

@media (max-width:800px) {
  .c13 {
    border-top: rgba(15,20,35,0.2) 1.4px dashed;
    padding-top: 1em;
    margin-top: 2em;
    border-bottom: none;
    padding-bottom: 0;
    margin-bottom: 0;
  }
}

@media (min-width:801px) {
  .c12 {
    margin-bottom: 1.5em;
  }
}

<div>
  <div
    class="c0 Card-hurxuz-0 c1 providerCard"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c0 c4"
        >
          <span
            class="c5"
            role="note"
          >
            <img
              alt="video icon"
              class="c6"
              src="ic_video.svg"
            />
            Atende Online
          </span>
        </div>
        <p
          class="c7"
          color="#ED985A"
        >
          Fisioterapia
        </p>
      </div>
      <div
        class="c8 providerCardColumn"
      >
        <p
          class="c9"
        >
          Thaís Amorim
        </p>
        <p
          class="c10"
        >
          Traumato-ortopedia
        </p>
        <div
          class="c11"
        >
          <p
            class="title"
          >
            REGISTRO
          </p>
          <p
            class="number"
          >
            Crefito4 202323
          </p>
        </div>
        <p
          class="c12"
        >
          Fisioterapeuta - Traumato Ortopedia
        </p>
        <div
          class="c13"
        >
          <p
            class="c14"
          >
            <img
              alt="address icon"
              class="c6"
              src="ic_pin.svg"
            />
            
            Minas Gerais (MG)
          </p>
          <p
            class="c15"
          >
            R. Bernardo Cupertino, 423 - B. Martins
          </p>
        </div>
      </div>
      <div
        class="c16 providerCardColumn"
      >
        <div
          class="c13"
        >
          <p
            class="c14"
          >
            Contato
          </p>
          <p>
            <img
              alt="phone icon"
              class="c6"
              src="ic_phone.svg"
            />
            <span>
               
              34 999624855
            </span>
          </p>
          <p
            class="c15"
          >
            <img
              alt="e-mail icon"
              class="c6"
              src="ic_mail.svg"
            />
            <span>
               
              thais_amorim773@hotmail.com
            </span>
          </p>
        </div>
        <div
          class="c13"
        >
          <p
            class="c14"
          >
            Atendimento online
          </p>
          <p>
            <img
              alt="video icon"
              class="c6"
              src="ic_video.svg"
            />
            <span>
              Zoom
            </span>
          </p>
        </div>
        <div
          class="c13"
        >
          <p
            class="c14"
          >
            Planos de saúde
          </p>
          <p>
            <img
              alt="card icon"
              class="c6"
              src="ic_card.svg"
            />
            <span>
              Sul América, IPSM, Saúde Caixa, Cassi, Amil e Vitalis
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProviderCard tests when mobile Component should render correctly 1`] = `
.c22 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #5996f7;
  stroke: #5996f7;
}

.c22 g {
  fill: inherit;
  stroke: inherit;
}

.c22 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c22 *[stroke*="#"],
.c22 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c22 *[fill-rule],
.c22 *[FILL-RULE],
.c22 *[fill*="#"],
.c22 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: header;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: first;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c18 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c18:hover {
  background-color: rgba(221,221,221,0.4);
  color: #000000;
}

.c18:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c18:focus > circle,
.c18:focus > ellipse,
.c18:focus > line,
.c18:focus > path,
.c18:focus > polygon,
.c18:focus > polyline,
.c18:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c18:focus::-moz-focus-inner {
  border: 0;
}

.c16 {
  -webkit-transition: max-height 200ms,opacity 200ms;
  transition: max-height 200ms,opacity 200ms;
  opacity: 0;
  overflow: hidden;
  max-height: 0;
}

.c21 {
  font-size: 14px;
  line-height: 20px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "header" "first" "second";
  grid-template-columns: auto;
  grid-row-gap: 0px;
  grid-column-gap: 0px;
  justify-items: stretch;
  grid-template-rows: auto auto auto;
}

.c15 {
  font-size: 14px;
  line-height: 20px;
  max-width: 336px;
}

.c1 {
  font-size: 10px;
  border-radius: 15px;
  box-shadow: 0px 10px 32px #0000001a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
  overflow: hidden;
  padding: 2em;
  line-height: 3em;
  word-break: break-word;
}

.c1 .providerCardColumn:last-child {
  border-right: none;
  margin-right: none;
}

.c10 {
  color: #0f1423;
  font-size: 1.4em;
  max-width: 30em;
  line-height: 20px;
  margin-top: 1em;
}

.c4 {
  width: 100%;
}

.c5 {
  color: #5996f7;
  background-color: #c7ddfd;
  border-radius: 0px 15px;
  font-size: 1.4em;
  font-weight: 600;
  padding: 0.5em 1em;
  position: absolute;
  right: 0;
  top: 0;
}

.c9 {
  font-weight: 600;
  color: #0f1423;
  font-size: 1.8em;
}

.c13 {
  border-bottom: #f5f5f5 1.5px solid;
  padding-bottom: 2em;
  margin-bottom: 2em;
  font-size: 1.4em;
}

.c13:last-child {
  border-bottom: none;
  padding-bottom: 0;
  margin-bottom: 0;
}

.c14 {
  font-weight: 600;
  color: #0f1423;
}

.c11 {
  margin-bottom: 1.5em;
  margin-top: 1.5em;
  line-height: 20px;
}

.c11 .title {
  font-weight: 800;
  color: #5996f7;
  font-size: 1.2em;
}

.c11 .number {
  color: #0f1423;
  font-size: 1.4em;
}

.c12 {
  font-size: 1.3em;
  line-height: 1.5em;
  color: #0f1423;
  opacity: 0.57;
}

.c6 {
  vertical-align: middle;
  margin-right: 5px;
}

.c7 {
  margin-right: 1em;
  color: #fff;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 70%;
  font-size: 1.4em;
  font-weight: 600;
  padding: 0.2em 2em;
  border-radius: 1.5em;
  line-height: 1.8em;
  background-color: #ED985A;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1.5em;
}

.c19 {
  font-size: 1.4rem;
  font-weight: 600;
  border-radius: 17px;
}

.c20 {
  background-color: #C7DDFD;
  color: #5996f7;
  width: 100%;
  height: 100%;
  padding: 20px 0 20px 0;
}

.c20 svg {
  vertical-align: middle;
}

.c17 {
  width: 100%;
  background-color: #C7DDFD;
  position: absolute;
  bottom: 0;
  left: 0;
  text-align: center;
}

@media (min-width:800px) {
  .c1 .providerCardColumn {
    border-right: #f5f5f5 1.5px dashed;
    padding-right: 3em;
  }

  .c1 .providerCardColumn:last-child {
    padding-left: 3em;
  }
}

@media (max-width:800px) {
  .c1 {
    padding-bottom: 10em;
  }
}

@media (max-width:440px) {
  .c4 {
    margin-top: 40px;
  }
}

@media (max-width:800px) {
  .c13 {
    border-top: rgba(15,20,35,0.2) 1.4px dashed;
    padding-top: 1em;
    margin-top: 2em;
    border-bottom: none;
    padding-bottom: 0;
    margin-bottom: 0;
  }
}

@media (min-width:801px) {
  .c12 {
    margin-bottom: 1.5em;
  }
}

<div>
  <div
    class="c0 Card-hurxuz-0 c1 providerCard"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c0 c4"
        >
          <span
            class="c5"
            role="note"
          >
            <img
              alt="video icon"
              class="c6"
              src="ic_video.svg"
            />
            Atende Online
          </span>
        </div>
        <p
          class="c7"
          color="#ED985A"
        >
          Fisioterapia
        </p>
      </div>
      <div
        class="c8 providerCardColumn"
      >
        <p
          class="c9"
        >
          Thaís Amorim
        </p>
        <p
          class="c10"
        >
          Traumato-ortopedia
        </p>
        <div
          class="c11"
        >
          <p
            class="title"
          >
            REGISTRO
          </p>
          <p
            class="number"
          >
            Crefito4 202323
          </p>
        </div>
        <p
          class="c12"
        >
          Fisioterapeuta - Traumato Ortopedia
        </p>
        <div
          class="c13"
        >
          <p
            class="c14"
          >
            <img
              alt="address icon"
              class="c6"
              src="ic_pin.svg"
            />
            
            Minas Gerais (MG)
          </p>
          <p
            class="c15"
          >
            R. Bernardo Cupertino, 423 - B. Martins
          </p>
        </div>
        <div
          class="c13"
        >
          <p
            class="c14"
          >
            Contato
          </p>
          <p>
            <img
              alt="phone icon"
              class="c6"
              src="ic_phone.svg"
            />
            <span>
               
              34 999624855
            </span>
          </p>
          <p
            class="c15"
          >
            <img
              alt="e-mail icon"
              class="c6"
              src="ic_mail.svg"
            />
            <span>
               
              thais_amorim773@hotmail.com
            </span>
          </p>
        </div>
        <div
          aria-hidden="true"
          class="c0 c16"
        >
          <div
            class="c13"
          >
            <p
              class="c14"
            >
              Atendimento online
            </p>
            <p>
              <img
                alt="video icon"
                class="c6"
                src="ic_video.svg"
              />
              <span>
                Zoom
              </span>
            </p>
          </div>
          <div
            class="c13"
          >
            <p
              class="c14"
            >
              Planos de saúde
            </p>
            <p>
              <img
                alt="card icon"
                class="c6"
                src="ic_card.svg"
              />
              <span>
                Sul América, IPSM, Saúde Caixa, Cassi, Amil e Vitalis
              </span>
            </p>
          </div>
        </div>
        <h2
          class="c17"
        >
          <button
            class="c18 c19 c20"
            type="button"
          >
            <span
              class="c21"
            >
              Saiba Mais
              <svg
                aria-label="FormDown"
                class="c22"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </span>
          </button>
        </h2>
      </div>
    </div>
  </div>
</div>
`;
